compiler/rustc_middle/src/ty/consts.rs

```diff
@@ -18,18 +20,28 @@ pub use kind::*;
 pub use valtree::*;
 
 /// Typed constant value.
-#[derive(Copy, Clone, Debug, Hash, TyEncodable, TyDecodable, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Copy, Clone, Debug, TyEncodable, TyDecodable, PartialEq, Eq, Ord, PartialOrd)]
 #[derive(HashStable)]
 pub struct Const<'tcx> {
-    pub ty: Ty<'tcx>,
+    pub(super) ty: Ty<'tcx>,
 
-    pub val: ConstKind<'tcx>,
+    pub(super) val: ConstKind<'tcx>,
```

Comment:
Please add a comment that the fields have restricted visibility because we want to prevent `Const` from being constructed in a non-interned way.